src/handler.py

```diff
@@ -154,8 +164,35 @@ def control_db_utilization(event, context):
         However, we know there is a backlog to work through so we need to force the db to maximum size
         by issuing a fake high-cpu alarm.
         """
-        logger.info(f"os.environ {os.environ}")
-        execute_recover_machine({}, {})
+        logger.info(f"The circuit breaker has calmed down.  Let's try to ramp things up.")
+        flow_rate = get_flow_rate()
+        if flow_rate == 0:
+            logger.info(f"Adjusting flow rate up to 15.")
+            adjust_flow_rate(15)
+        elif flow_rate == 15:
+            logger.info(f"Adjusting flow rate up to 25.")
+            adjust_flow_rate(25)
+        elif flow_rate == 25:
+            logger.info(f"We already at maximum flow rate, so no change made.")
```

Comment:
I'm unfamiliar with how cloudwatch events fire off. If we go from `ALARM` to `OK`, is that it? If it doesn't trigger again, we might get stuck at 15. Is that a scenario that can happen?

Reply:
It transitions a lot between OK and INSUFFICIENT_DATA.  If you look at the logs right now, you'll see a lot of "flow_rate is already at maximize so don't make any changes" type of messages.

It could potentially get stuck at 15 for a period of time because -- using an alarm -- we're subject to the vagaries of how AWS does OK and INSUFFICIENT_DATA.   I don't see how to avoid that possibility if we are trying to ramp up and down and using alarms, though.

Reply:
I can live with that. If we really wanted to we could add a function or capability to the troubleshoot function to adjust concurrency. I actually have that in another branch... :thinking: